torch/_dynamo/variables/higher_order_ops.py

```diff
@@ -824,10 +826,16 @@ def diff_meta(tensor_vars1, tensor_vars2):
             true_shared + unique_true + unique_false,
         )
 
-        flat_example_value = pytree.tree_map_only(
-            torch.fx.Proxy,
-            lambda a: a.node.meta["example_value"],
+        from torch._higher_order_ops.cond import _merge_tensors
+
+        flat_example_value = pytree.tree_map(
+            lambda a, b: _merge_tensors(
+                a.node.meta["example_value"], b.node.meta["example_value"], tx.fake_mode
+            )
+            if isinstance(a, torch.fx.Proxy) and isinstance(b, torch.fx.Proxy)
```

Comment:
So, it's kind of annoying that the differencing code and then merging has to be implemented multiple places. In an ideal world, we would run fake tensor prop through torch.cond and that would give us the final example values here.